### PR TITLE
Fix readme and add changelog for release-2.4

### PR DESCRIPTION
**Pull request by shiyuhang0**
### What problem does this PR solve? 
release-2.4's doc need update.
- update readme
- add changes in changelog

### Check List 

Tests 

 - No code

### CODE REVIEW

**Comment by shiyuhang0** [comment]
PTAL @sunxiaoguang @xuanyu66 @zhangyangyu

File: README.md [hunk 1]
```diff
@@ -111,7 +75,7 @@ The figure below show the architecture of TiSpark.
 
 ![architecture](./docs/architecture.png)
 
-+ TiSpark integrates well with the Spark Catalyst Engine. It provides precise control of computing, which allows Spark to read data from TiKV efficiently. It also supports index seek, which significantly improves the performance of the point query execution.
++ TiSpark integrates well with the Spark Catalyst Engine. It provides precise control of comfputing, which allows Spark to read data from TiKV efficiently. It also supports index seek, which significantly improves the performance of the point query execution.
```
Comment (zhangyangyu) [inline README.md:78]: ? seems not right

Comment (shiyuhang0) [reply README.md:78]: coopy from matser, i think both are wrong

**Review by sunxiaoguang — APPROVED** [review]
LGTM

**Review by sunxiaoguang — APPROVED** [review]

**Pull request merged**